src/H5private.h

```diff
@@ -1394,6 +1394,13 @@ H5_DLL H5_ATTR_CONST int Nflock(int fd, int operation);
 #ifndef HDstrcasecmp
 #define HDstrcasecmp(X, Y) strcasecmp(X, Y)
 #endif
+#ifndef HDstrcasestr
+#ifdef _WIN32
+#define HDstrcasestr(X, Y) StrStrIA(X, Y)
+#else
+#define HDstrcasestr(X, Y) strcasestr(X, Y)
```

Comment:
Hey @berghaus, two things here:

- The #define HDstrcasestr for Windows should be moved into H5win32defs.h since that's where all the Windows compatibility stuff goes

- Though it's been around long enough that there probably aren't many (or any) systems where we'd need to worry about having strcasestr, it would probably be better and more portable for HDstrcasestr to point to an implemented function, say H5strcasestr, so we guarantee we have the functionality across platforms.

Reply:
Thank you for the feedback @jhendersonHDF :-)

Regarding your second point: I had been considering this as well - but ran into trouble when dealing with localization. Here would be a possible solution, but I am unsure where to best put this into the existing code [1]:
```c
char *
H5strcasestr(const char *haystack, const char* needle)
{
    /* Check arguments. */
    HDassert(haystack);
    HDassert(needle);

    /* begin once from each character of haystack, until needle is found */
    do {
        const char *h = haystack;
        const char *n = needle;
        /* loop while lowercase strings match, or needle ends */
        while (HDtolower(*h) == HDtolower(*n) && *n) {
            h++;
            n++;
        }
        /* if all characters in needle matched we found it */
        if (*n == 0) {
            return (char *)haystack;
        }
    } while (*haystack++);
    return 0;
} /* end H5strcasestr() */
```

I am a bit uncertain where to best place this, and how to decorate it. I was trying to follow what is done with `H5_timer_get_time_string` as an example - but I still get linking errors when trying. Shall I mark this as a draft and push this code, so you can more concretely see what I am doing wrong?

[1] Inspired by [this discussion](https://stackoverflow.com/questions/27303062/strstr-function-like-that-ignores-upper-or-lower-case).

Reply:
> I am a bit uncertain where to best place this

Though not the most ideal place, I'd say the best candidate currently would be for this to go into H5system.c and then the prototype for the function would go in H5private.h towards the bottom (making sure that the prototype is marked with H5_DLL).

> Shall I mark this as a draft and push this code, so you can more concretely see what I am doing wrong?

Sure, it never hurts to have the current code here to view! Your implementation looks fine to me though, so I guess we just need to figure out why you end up with linking errors.

Reply:
> Sure, it never hurts to have the current code here to view! Your implementation looks fine to me though, so I guess we just need to figure out why you end up with linking errors.

I made a mistake, trying to add a new source file in the make instructions. Following your suggestion, the linking errors went away :-)